src/sentry/utils/query.py

```diff
@@ -192,26 +210,25 @@ def __init__(self, iterator, count=None, caption=None):
         self.caption = str(caption or "Progress")
 
     def __iter__(self):
-        if self.count != 0:
```

Comment:
Tests were failing because postgres doesn't update the stats in time for them to work in our tests. This doesn't seem to have any real impact on how these run, and I added a test to cover the empty queryset path.